test/jdk/javax/swing/plaf/nimbus/TestNimbusLabel.java

```diff
@@ -0,0 +1,120 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/*
+ * @test
+ * @bug 7175396
+ * @key headful
+ * @summary  Verifies the text on label is painted red for Nimbus LaF.
+ * @run main/othervm -Dsun.java2d.uiScale=1.0 TestNimbusLabel
+ */
+
+import java.io.File;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+import javax.imageio.ImageIO;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+public class TestNimbusLabel
+{
+    private static JLabel label;
+    private static JFrame frame;
+    private static boolean passed = false;
+    private static int colorTolerance = 5;
+
+    private static boolean checkPixel(Color color) {
+        int red1 = color.getRed();
+        int blue1 = color.getBlue();
+        int green1 = color.getGreen();
+        int red2 = Color.red.getRed();
+        int blue2 = Color.red.getBlue();
+        int green2 = Color.red.getGreen();
+        if ((Math.abs(red1 - red2) < colorTolerance) &&
+                (Math.abs(green1 - green2) < colorTolerance) &&
+                (Math.abs(blue1 - blue2) < colorTolerance)) {
+            return true;
+        }
+        return false;
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        try {
+            UIManager.setLookAndFeel("javax.swing.plaf.nimbus.NimbusLookAndFeel");
+            Robot robot = new Robot();
+
+            SwingUtilities.invokeAndWait(() -> {
+                frame = new JFrame();
+                UIManager.getDefaults().put("Label.foreground",
+                                            java.awt.Color.red);
+                label =
+                    new JLabel("<html><body>This text should be in red</body></html>");
+
+                frame.getContentPane().add(label);
+                frame.setUndecorated(true);
+                frame.setLocationRelativeTo(null);
+                frame.pack();
+                frame.setVisible(true);
+            });
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            Point p = frame.getLocationOnScreen();
+            Dimension d = frame.getSize();
+            System.out.println("width " + d.getWidth() +
+                                " height " + d.getHeight());
+            int y = p.y + d.height/2;
+
+            for (int x = p.x; x < p.x + d.width; x++) {
+                    System.out.println("color(" + x + "," + y + ")=" +
+                                        robot.getPixelColor(x, y));
+                    Color color = robot.getPixelColor(x, y);
+                    if (checkPixel(color)) {
```

Comment:
I wonder if disabling text antialiasing would yield a simpler code in `checkPixel`? However, it cannot be disabled on macOS which still uses grayscale antialiasing. But for Windows and Linux, it should help. Or the test could be limited to Windows and Linux.